devfiles/rust/devfile.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: 1.0.0
+metadata:
+  generateName: rust-
+projects:
+  - name: HelloWorld-Rust
+    source:
+      location: 'https://github.com/dnnr/HelloWorld-Rust.git'
```

Comment:
hello, maybe https://github.com/rust-lang/rust-by-example repository might be more mainstream with subfolder `src/hello` ?

Reply:
Yeah, I had the same process :)

Reply:
> I'm not 100% sure on the best approach for a sample project, even though the one used is a minimal example. Some builds of the devfile registry redistribute code, and unlicensed code can be strange.

I'm not overly happy with this either, but I couldn't find any other good (ideally official) sample project. I think at the very least, a trivial sample (like that one) should be added to `github.com/che-samples`, but I don't know the contribution process there. If this can be done soonish, I'd update by PR accordingly.

Reply:
Turns out I have the required permissions :)

Created https://github.com/che-samples/helloworld-rust (includes an updated devfile)

I haven't added a license here, since I'm not sure our policy for Che samples (many use MIT, but I feel like EPL may be more appropriate).

Reply:
I also went ahead and created https://quay.io/repository/eclipse/che-rust-1.39 to host the new image.

Reply:
Thanks, I've updated this PR.

@amisevsk Are you sure that the devfile is supposed to be part of the sample repository? I doesn't seem to be the case in the other samples.

Reply:
It's not a requirement, but it can be a nice introduction to Che and how it can be used; for example, most of the Che components have a devfile included in the repo, and this allows users to quickly start a workspace from that repo, regardless of the devfile registry. 

There is something to be said for having to keep the two devfiles in sync though. I'll have to think about that.